enginetest/enginetests.go

```diff
@@ -5401,22 +5401,54 @@ func TestAlterTable(t *testing.T, harness Harness) {
 		}, checks)
 	})
 
-	t.Run("drop column preserves check constraints", func(t *testing.T) {
+	t.Run("drop column drops check constraint", func(t *testing.T) {
 		RunQuery(t, e, harness, "create table t34 (i bigint primary key, s varchar(20))")
-		RunQuery(t, e, harness, "ALTER TABLE t34 ADD COLUMN j int, ADD COLUMN k int")
+		RunQuery(t, e, harness, "ALTER TABLE t34 ADD COLUMN j int")
 		RunQuery(t, e, harness, "ALTER TABLE t34 ADD CONSTRAINT test_check CHECK (j < 12345)")
-
-		AssertErr(t, e, harness, "ALTER TABLE t34 DROP COLUMN j", sql.ErrCheckConstraintInvalidatedByColumnAlter)
-
-		RunQuery(t, e, harness, "ALTER TABLE t34 DROP COLUMN k")
+		RunQuery(t, e, harness, "ALTER TABLE t34 DROP COLUMN j")
 		tt := queries.QueryTest{
 			Query: "show create table t34",
 			Expected: []sql.Row{{"t34", "CREATE TABLE `t34` (\n" +
 				"  `i` bigint NOT NULL,\n" +
 				"  `s` varchar(20),\n" +
-				"  `j` int,\n" +
+				"  PRIMARY KEY (`i`)\n" +
+				") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"}},
+		}
+		TestQueryWithEngine(t, harness, e, tt)
+	})
+
+	t.Run("drop column drops all relevant check constraints", func(t *testing.T) {
+		RunQuery(t, e, harness, "create table t42 (i bigint primary key, s varchar(20))")
+		RunQuery(t, e, harness, "ALTER TABLE t42 ADD COLUMN j int")
+		RunQuery(t, e, harness, "ALTER TABLE t42 ADD CONSTRAINT check1 CHECK (j < 12345)")
+		RunQuery(t, e, harness, "ALTER TABLE t42 ADD CONSTRAINT check2 CHECK (j > 0)")
+		RunQuery(t, e, harness, "ALTER TABLE t42 DROP COLUMN j")
+		tt := queries.QueryTest{
+			Query: "show create table t42",
+			Expected: []sql.Row{{"t42", "CREATE TABLE `t42` (\n" +
+				"  `i` bigint NOT NULL,\n" +
+				"  `s` varchar(20),\n" +
+				"  PRIMARY KEY (`i`)\n" +
+				") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"}},
+		}
+		TestQueryWithEngine(t, harness, e, tt)
+	})
+
+	t.Run("drop column drops correct check constraint", func(t *testing.T) {
```

Comment:
Would be nice to add one more test here that asserts that a constraint with multiple columns won't be deleted. I see that you have those in the PreparedScriptTests, but would be nice to have something in `TestAlterTable`, too.